File: rust/src/models/onboarding_link_response.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// Result of creating an onboarding link
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct OnboardingLinkResponse {
    pub expires_at: String,

    pub url: String,
}

impl OnboardingLinkResponse {
    pub fn new(expires_at: String, url: String) -> Self {
        Self { expires_at, url }
    }
}

File: typescript/src/models/onboardingLinkResponse.ts
// this file is @generated
import { parseDateTime } from "../datetime";
/** Result of creating an onboarding link */
export interface OnboardingLinkResponse {
  expiresAt: Date;

  url: string;
}

export const OnboardingLinkResponseSerializer = {
  _fromJsonObject(object: any): OnboardingLinkResponse {
    return {
      expiresAt: parseDateTime(object["expires_at"]),
      url: object["url"],
    };
  },

  _toJsonObject(self: OnboardingLinkResponse): any {
    return {
      expires_at: self.expiresAt,
      url: self.url,
    };
  },
};
